npdarrington/fitlit
# Nd sleep class

What’s this PR do?  
- This PR includes the Sleep class as well as the testing suite Sleep-test.js.
- This PR will allow us to call specific methods that will display information to the user's dashboard.
- A user will be able to see their average number of hours slept, sorted by one day or multiple day increments.
- A user will be able to see their average sleep quality, sorted by one day or multiple day increments.
- A user will be able to see all users sleep quality as an average and all users over 3 sleep quality between specific dates.
- A user will be able to see who slept the most number of hours for a specific day.
 
Where should the reviewer start?  
- Git pull this branch.
 
How should this be manually tested?  
- Run `npm test` to test Sleep.js against the Sleep-test.js
- Check for context, workflow, SRP and DRY.
 
Any background context you want to provide?  
- Displaying to the DOM will be on a future branch.
- This code is not DRY or SRP at all. Helper functions will be added. Pushed up for tonight to save changes. While all of the functionality works, this definitely needs to and will be refactored.
 
What are the relevant tickets?  
Closes #28 , Closes #23, Closes #17, Closes #16, Closes #13, Closes #15, Closes #8, Closes #11, Closes #10, Closes #19, Closes #7, Closes #25, 
 
Screenshots (if appropriate)  
- N/A
 
Questions: 
- N/A
